### PR TITLE
Updates for Dart 2.0 core library changes (wave 2.2)

### PR DESCRIPTION
**Pull request by leafpetersen**
Change classes that implement `StreamTransformer` to extend `StreamTransformerBase`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+## 0.9.0+4
+
+* Updates to support Dart 2.0 core library changes (wave
+  2.2). See [issue 31847][sdk#31847] for details.
+  
+  [sdk#31847]: https://github.com/dart-lang/sdk/issues/31847
+
```
Comment (kevmoo) [inline CHANGELOG.md]: extra lines here

**Review by kevmoo — APPROVED** [review]
No travis support, which is sad.

Did you verify the tests?

**Comment by leafpetersen** [comment]
tests pass locally.

File: CHANGELOG.md (hunk 1 above)
Comment (leafpetersen) [reply CHANGELOG.md]: done.

**Comment by leafpetersen** [comment]
@kevmoo will handle publishing

**Pull request merged**